src/Model/Connection/GridSorting.f90

```diff
@@ -0,0 +1,84 @@
+module GridSorting  
+  use KindModule, only: I4B, DP, LGP
+  use ConstantsModule, only: DHALF
+  use TopologyModule, only: GlobalCellType
+  use GenericUtilitiesModule, only: is_same
+  implicit none
+  private
+
+  public :: quickSortGrid
+
+contains
+  ! Sort an array of integers
+  subroutine quickSortGrid(array, arraySize, idxToGlobal)
+      integer, intent(inout), dimension(:) :: array      
```

Comment:
should standard MODFLOW 6 datatypes (I4B, etc) be used here and in `qsort_inline.inc`?